src/main/java/com/elixr/poc/service/PurchaseService.java

```diff
@@ -20,17 +22,34 @@ public PurchaseService(PurchaseRepository purchaseRepository) {
      *
      * @param purchaseId
      * @return
-     * @throws NoRecordFoundException
+     * @throws GlobalException
      */
-    public boolean deletePurchaseDetails(UUID purchaseId) throws NoRecordFoundException {
-        boolean success = false;
+    public boolean deletePurchaseDetails(UUID purchaseId) throws GlobalException {
+        boolean success;
         boolean purchaseRecordExists = purchaseRepository.existsById(purchaseId);
         if (purchaseRecordExists) {
             purchaseRepository.deleteById(purchaseId);
             success = true;
         } else {
-            throw new NoRecordFoundException(ApplicationConstants.ID_MISMATCH);
+            throw new GlobalException(ApplicationConstants.ID_MISMATCH);
```

Comment:
Error message not matching with the specific use case. Here the scenario is the id does not exist, not the id mismatch.

Reply:
Updated